code/model/UserDefinedForm.php

```diff
@@ -459,7 +459,8 @@ public function ping()
      */
     public function Form()
     {
-        $form = UserForm::create($this);
+        $form = UserForm::create($this, 'Form_' . $this->ID);
+        $form->setFormAction(Controller::join_links($this->owner->Link(), 'Form'));
```

Comment:
UserDefinedForm isn't an extension; `$this->owner->Link()` should just be `$this->Link()`